src/OMSimulatorLib/TLMBusConnector.cpp

```diff
@@ -260,6 +305,48 @@ oms3::Component *oms3::TLMBusConnector::getComponent()
   return component;
 }
 
+oms3::TLMBusConnector* oms3::TLMBusConnector::getActualBus()
+{
+  if(!actualBus && !parentSystem)
+    actualBus = this;
+  else if(!actualBus)
+    actualBus = getActualBus(getName(), parentSystem);
+
+  return actualBus;
+}
+
+oms3::TLMBusConnector* oms3::TLMBusConnector::getActualBus(ComRef cref, System *system)
+{
+  Connection** connections = system->getConnections(ComRef(""));
+  for(int i=0; connections[i]; ++i) {
+    ComRef subcref = ComRef("");
+    if(connections[i]->getSignalA() == cref) {
+      subcref = connections[i]->getSignalB();
+    }
+    else if(connections[i]->getSignalB() == cref) {
+      subcref = connections[i]->getSignalA();
+    }
+    else {
+      return this;
+    }
+
+    TLMBusConnector* subBus = system->getTLMBusConnector(subcref);
```

Comment:
And here as well: What does `system->getTLMBusConnector(subcref)` if `subref` is an empty `ComRef`?

Reply:
"subcref" will never be empty, due to the if statement above. It will either assign subcref with something or return a self-pointer. I only assign subcref with an empty string because I want to create it before the if statement and the constructor requires a string argument.